impl/maven-impl/src/test/java/org/apache/maven/impl/DefaultModelXmlFactoryTest.java

```diff
@@ -122,4 +126,91 @@ void testMalformedModelVersion() throws Exception {
         assertEquals("invalid.version", model.getModelVersion());
     }
+
+    @Test
+    void testWriteModelWithoutInputLocationTracking() {
+        String model =
+                """
+                <project xmlns="http://maven.apache.org/POM/4.0.0">
+                  <modelVersion>4.0.0</modelVersion>
+                    <groupId>org.example</groupId>
+                    <artifactId>example-artifact</artifactId>
+                    <version>1.0.0</version>
+                    <name>Example Project</name>
+                    <properties>
+                        <example.property>value</example.property>
+                    </properties>
+                </project>""";
+
+        String expected =
+                """
+                <?xml version="1.0" encoding="UTF-8"?>
+                <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+                  <modelVersion>4.0.0</modelVersion>
+                  <groupId>org.example</groupId>
+                  <artifactId>example-artifact</artifactId>
+                  <version>1.0.0</version>
+                  <name>Example Project</name>
+                  <properties>
+                    <example.property>value</example.property>
+                  </properties>
+                </project>""";
+
+        final StringWriter writer = new StringWriter();
+        XmlWriterRequest<Model> request = XmlWriterRequest.<Model>builder()
+                .content(factory.read(XmlReaderRequest.builder()
+                        .reader(new StringReader(model))
+                        .build()))
+                .writer(writer)
+                .build();
+
+        factory.write(request);
+
+        final String processedPomAsString = writer.toString();
+        assertThat(processedPomAsString).isEqualTo(expected);
+    }
+
+    @Test
+    void testWriteModelWithInputLocationTracking() {
+        String model =
+                """
+                <project xmlns="http://maven.apache.org/POM/4.0.0">
+                  <modelVersion>4.0.0</modelVersion>
+                    <groupId>org.example</groupId>
+                    <artifactId>example-artifact</artifactId>
+                    <version>1.0.0</version>
+                    <name>Example Project</name>
+                    <properties>
+                        <example.property>value</example.property>
+                    </properties>
+                </project>""";
+
+        String expected =
+                """
+                <?xml version="1.0" encoding="UTF-8"?>
+                <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+                  <modelVersion>4.0.0</modelVersion><!--some formatter: n/a @ 2:3-->
```

Comment:
I do not understand why someone would want this in their output. It feels very special case. This is not what I thought location information was, and I'm tempted to take this out.

Reply:
It seems like something that is valid be discussed, but I suggest to raise a new issue/discussion on this as it won't be fixed in this PR. 

This is the current implementation on location tracking when a formatter is present. This PR changes _when_ it should trigger the formatting, not _how_ it is formatted. Whether it should be parsed as a new element, comment, attribute, or something else is not part of this change.